### PR TITLE
Use v-model for inputs

### PR DESCRIPTION
**Pull request by FuriouZz**
Use v-model logic.
Issue #190

---
This change is [Reviewable](https://reviewable.io/reviews/graphiteeditor/graphite/224)

### CODE REVIEW

**Review by Keavon — CHANGES_REQUESTED** [review]
Thanks for this! Would you also please also add this `v-model` reactivity to `NumberInput.vue`? We will need to make use of the number input widget very soon for a couple new features that the backend is building. This should probably be updated whenever the user hits enter (which should remove focus from text entry) or when the user clicks out of the input box (and therefore removes focus from the text entry).

> *[Reviewable](https://reviewable.io/reviews/graphiteeditor/graphite/224#-:-MccYLfn46JdSKyuQrgW:b42xg6x)* status: 0 of 5 files reviewed, 2 unresolved discussions (waiting on @FuriouZz)

___
*[client/web/src/components/widgets/floating-menus/FloatingMenu.vue, line 354 at r1](https://reviewable.io/reviews/graphiteeditor/graphite/224#-MccVqmNAx8Fu82sb_LG:-MccVqmNAx8Fu82sb_LH:b-dl1tie) ([raw file](https://github.com/graphiteeditor/graphite/blob/4870da3f910d7421d2d6eb1a9988c839ced523ea/client/web/src/components/widgets/floating-menus/FloatingMenu.vue#L354)):*
> ```Vue
> 	},
> 	computed: {
> 		floatingMenuContentStyle(): unknown {
> ```

Thanks for this change! It also reminded me that you made your excellent feedback on my PR shortly after I submitted it, and I don't think I addressed that. I'm going to open it up so I don't forget again.

Is it possible to avoid the `unknown` type? For example, can we use a `{ minWidth: string }` type, or define an interface?
___
*[client/web/src/components/widgets/inputs/DropdownInput.vue, line 8 at r1](https://reviewable.io/reviews/graphiteeditor/graphite/224#-MccXeZGBUWD-LLZs08j:-MccXeZGBUWD-LLZs08k:bj3rakz) ([raw file](https://github.com/graphiteeditor/graphite/blob/4870da3f910d7421d2d6eb1a9988c839ced523ea/client/web/src/components/widgets/inputs/DropdownInput.vue#L8)):*
> ```Vue
> 			<Icon :class="'dropdown-arrow'" :icon="'DropdownArrow'" />
> 		</div>
> 		<MenuList :menuEntries="menuEntries" v-model:active-entry="activeEntry" :direction="MenuDirection.Bottom" @width-changed="onWidthChanged" :drawIcon="drawIcon" ref="menuList" />
> ```

Question (not requesting a change): what should determine whether a property is camelCase or kebab-case? I see you changed `activeEntry` to `v-model:active-entry` and `:widthChanged` to `@width-changed`. Is it everything except `v-bind:foo` (shorthand `:foo`) items? I want to learn the pattern for what I presume is the best practice in Vue so I can follow it from now on if I can remember to.

**Comment by FuriouZz** [comment]
> Thanks for this! Would you also please also add this `v-model` reactivity to `NumberInput.vue`? We will need to make use of the number input widget very soon for a couple new features that the backend is building. This should probably be updated whenever the user hits enter (which should remove focus from text entry) or when the user clicks out of the input box (and therefore removes focus from the text entry).

You right, I must watch on every inputs. We have to see every input as a HTML Input and use v-model on it.

> Thanks for this change! It also reminded me that you made your excellent feedback on my PR shortly after I submitted it, and I don't think I addressed that. I'm going to open it up so I don't forget again.
> 
> Is it possible to avoid the `unknown` type? For example, can we use a `{ minWidth: string }` type, or define an interface?

Of course

> Question (not requesting a change): what should determine whether a property is camelCase or kebab-case? I see you changed `activeEntry` to `v-model:active-entry` and `:widthChanged` to `@width-changed`. Is it everything except `v-bind:foo` (shorthand `:foo`) items? I want to learn the pattern for what I presume is the best practice in Vue so I can follow it from now on if I can remember to.

@Keavon I used kebab-case for props because this is a style recommandation from Vue ([link](https://vuejs.org/v2/style-guide/#Prop-name-casing-strongly-recommended)). In HTML, it is more common to use kebab-case, where as in Javascript camelCase. Vue do the translation for you, it's not a problem. If you were using JSX, camelCase for props would be prefered.

**Comment by FuriouZz** [comment]
I saw that we cannot use shortcuts for development case (opening inspector or refresh the tab). It could be cool to allow these shortcuts in development environment for ease.

**Review by Keavon — COMMENTED** [review]
Please rebase, I just fixed that! We can use F11 (fullscreen), F12 (open dev tools), Ctrl+Shift+C (select element in DOM), and type numbers into the text input fields again.

> *[Reviewable](https://reviewable.io/reviews/graphiteeditor/graphite/224#-:-McciJnHEWCa7AKZJ8FX:bhufogl)* status: 0 of 5 files reviewed, 2 unresolved discussions (waiting on @FuriouZz)

**Review by Keavon — COMMENTED** [review]
Does this also mean we should update all our `:fooBar` into `:foo-bar` throughout the app?

> *[Reviewable](https://reviewable.io/reviews/graphiteeditor/graphite/224#-:-MccicYd3EiEuH_CsCld:b-5tebk2)* status: 0 of 5 files reviewed, 2 unresolved discussions (waiting on @FuriouZz)

**Review by FuriouZz — COMMENTED** [review]
Yes, it will be better according to Vue Style Guide. It could be good that our frontend style guide stick to Vue Style Guide too.

> *[Reviewable](https://reviewable.io/reviews/graphiteeditor/graphite/224#-:-McdIbIg0SbiFoGH89eD:b-usrdf6)* status: 0 of 7 files reviewed, 2 unresolved discussions (waiting on @FuriouZz)

**Review by FuriouZz — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/graphiteeditor/graphite/224)* status: 0 of 7 files reviewed, 2 unresolved discussions (waiting on @FuriouZz and @Keavon)

___
*[client/web/src/components/widgets/floating-menus/FloatingMenu.vue, line 354 at r1](https://reviewable.io/reviews/graphiteeditor/graphite/224#-MccVqmNAx8Fu82sb_LG:-McdJ7RY6a0aC34YRR6D:bboa2g5) ([raw file](https://github.com/graphiteeditor/graphite/blob/4870da3f910d7421d2d6eb1a9988c839ced523ea/client/web/src/components/widgets/floating-menus/FloatingMenu.vue#L354)):*
<details><summary><i>Previously, Keavon (Keavon Chambers) wrote…</i></summary><blockquote>

Thanks for this change! It also reminded me that you made your excellent feedback on my PR shortly after I submitted it, and I don't think I addressed that. I'm going to open it up so I don't forget again.

Is it possible to avoid the `unknown` type? For example, can we use a `{ minWidth: string }` type, or define an interface?
</blockquote></details>

I used CSSStyleDeclaration

**Review by FuriouZz — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/graphiteeditor/graphite/224)* status: 0 of 7 files reviewed, 2 unresolved discussions (waiting on @FuriouZz and @Keavon)

___
*[client/web/src/components/widgets/floating-menus/FloatingMenu.vue, line 354 at r1](https://reviewable.io/reviews/graphiteeditor/graphite/224#-MccVqmNAx8Fu82sb_LG:-McdJNeR3BR4WlUfj7QE:b-896fix) ([raw file](https://github.com/graphiteeditor/graphite/blob/4870da3f910d7421d2d6eb1a9988c839ced523ea/client/web/src/components/widgets/floating-menus/FloatingMenu.vue#L354)):*
<details><summary><i>Previously, FuriouZz (Chrs Msln) wrote…</i></summary><blockquote>

I used CSSStyleDeclaration
</blockquote></details>

Done.

File: client/web/src/components/widgets/inputs/NumberInput.vue [hunk 1]
```diff
@@ -1,8 +1,8 @@
 <template>
```
Comment (Keavon) [inline client/web/src/components/widgets/inputs/NumberInput.vue:1]: Would you have time to modify this so it behaves such that: you click in the center (not the left or right arrow) and it puts you into text entry mode (while the <input> is focused) with a slightly lighter background color, and the unit is removed from the number, the left/right arrows are hidden, text entry is left-aligned, and all the text starts highlighted so it's easy to replace the whole number, and finally when you hit enter or click out of the component, it applies the number and restores the styling including the unit reappears, and the updated number is only then sent to the user of the component.

**Comment by Keavon** [comment]
Hi @FuriouZz, I just wanted to check that you saw my previous comment. If you don't have time to add that functionality, I can do it over the next few days. It'll be needed for setting opacity which @geom3trik is working on.

**Comment by Keavon** [comment]
I'll go ahead and merge this and one of us can implement the functionality I described above.

**Pull request merged**